contracts/PoolTest.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.6.2;
+pragma experimental ABIEncoderV2;
+
+import "./Pool.sol";
+
+contract ReceiverWeightsTest {
+    using ReceiverWeightsImpl for mapping(address => ReceiverWeight);
+
+    mapping(address => ReceiverWeight) private receiverWeights;
+    ReceiverWeightIterated[] private receiverWeightsIterated;
+    int256 public receiverWeightsSumDelta;
+
+    struct ReceiverWeightIterated {
+        address receiver;
+        uint32 weight;
+    }
+
+    function setWeights(ReceiverWeightIterated[] calldata weights) external {
+        int256 weightsSumDelta = 0;
+        for (uint256 i = 0; i < weights.length; i++) {
+            weightsSumDelta += weights[i].weight;
+            weightsSumDelta -= receiverWeights.setWeight(
+                weights[i].receiver,
+                weights[i].weight
+            );
+        }
+        receiverWeightsSumDelta = weightsSumDelta;
+        delete receiverWeightsIterated;
+        address receiver = address(0);
+        while (true) {
+            uint32 weight;
+            (receiver, weight) = receiverWeights.nextWeight(receiver);
+            if (weight == 0) break;
+            receiverWeightsIterated.push(
+                ReceiverWeightIterated(receiver, weight)
+            );
+        }
+    }
+
+    function getReceiverWeightsIterated()
+        external
+        view
+        returns (ReceiverWeightIterated[] memory)
+    {
+        return receiverWeightsIterated;
+    }
```

Comment:
if `receiverWeightsIterated` is `public`, it can be accessed directly via `contract.receiverWeightsIterated()`

Reply:
Not quite, `contract.receiverWeightsIterated()` requires an index parameter and returns a single item. That's inconvenient for testing.

Reply:
Ah gotcha.